src/frontend/download-chrome.ts

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
+import * as os from "os";
+import { promisify } from "util";
+import * as fs from "fs";
+import got from "got";
+import stream from "node:stream";
+import extract from "extract-zip";
+import { unreachable } from "./src/utils/utils";
+
+const pipeline = promisify(stream.pipeline);
+
+const stat = promisify(fs.stat);
+const unlink = promisify(fs.unlink);
+
+const installPath = `${os.homedir()}/.local-chromium`;
+
+type Config = { platform: "linux" | "darwin"; revision: string };
+
+/* This downloads Chromium from the Chromium build archives
+ * For more info on finding a particular version of chromium, see https://www.chromium.org/getting-involved/download-chromium/
+ * This also caches chromium downloads. The following happens:
+ * - check module cache
+ * -    if exists, return
+ * - check global cache
+ * -    if exists, copy and return
+ * - install into global cache
+ * - copy and return
+ *
+ * This is a simplified version of https://github.com/juliangruber/download-chromium, translated
+ * to TypeScript in the process.
+ */
+export async function downloadChrome() {
+  const config = getConfig();
+
+  // First, figure out if we have already downloaded chromium
+
+  const moduleExecutablePath = getExecutablePath(config, installPath);
+  try {
+    await stat(moduleExecutablePath);
+    return moduleExecutablePath;
+  } catch (_) {
+    /* noop */
+  }
+
+  // If not, create the cache and download
+
+  try {
+    fs.mkdirSync(installPath);
+  } catch (_) {
+    /* noop */
+  }
+  const folderPath = getFolderPath(config, installPath);
+  const zipPath = `${folderPath}.zip`;
+
+  await pipeline(
+    await got.stream(downloadURL(config)),
+    fs.createWriteStream(zipPath)
+  );
+
+  // Now extract the zip and install it to the cache
+  await extract(zipPath, { dir: folderPath });
+  await unlink(zipPath);
+
+  return moduleExecutablePath;
+}
+
+function getConfig(): Config {
+  const platform = os.platform();
+
+  if (platform !== "linux" && platform !== "darwin") {
+    throw new Error(`Platform ${platform} is not supported`);
+  }
+
+  // Follow steps here to update: https://www.chromium.org/getting-involved/download-chromium/
+  const revision = platform === "darwin" ? "1036822" : "1036826";
+
+  return { platform, revision };
+}
+
+/** The chromium download URL for the given platform and revision */
+function downloadURL(config: Config) {
+  if (config.platform === "darwin") {
```

Comment:
nth:

- you can deconstruct `{revision, platform}`
- `https://www.googleapis.com/storage/v1/b/chromium-browser-snapshots/o/` seems to be the same for the two urls